arch/arm/core/Kconfig

```diff
@@ -45,6 +45,19 @@ config ARM_STACK_PROTECTION
 menu "Architectue Floating Point Options"
 depends on CPU_HAS_FPU
 
+config ARM_SECURE_FIRMWARE
+	bool
+	default n
+	help
+	  This option indicates that we are building a Secure application for
```

Comment:
I am not sure about using "application" here. Maybe "image" is better? or will we later generate flash images that contain both secure and non-secure code?

Reply:
@carlescufi IMO, image should refer to the combination of secure and non secure applications that will share the mcu 

That's why I picked the "application"

Reply:
But will they phisically be an image, or two? I assume a single one, since there will be a single instance of the kernel and libraries?

Reply:
It will be a single image, but kernel or lib sharing depends on use case, IMO. I don't think we need to address this in this PR.

Reply:
it was my understanding that ARM was going to use a modified bootloader and use two separate images, one for secure and one for non-secure.

Reply:
@carlescufi Changed, "application" to "firmware". Taken directly from the ARM PSA overview documentation.